homeassistant/components/foscam/camera.py

```diff
@@ -28,20 +60,70 @@
     }
 )
 
+SERVICE_PTZ_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_MOVEMENT): vol.In(
+            [
+                DIR_UP,
+                DIR_DOWN,
+                DIR_LEFT,
+                DIR_RIGHT,
+                DIR_TOPLEFT,
+                DIR_TOPRIGHT,
+                DIR_BOTTOMLEFT,
+                DIR_BOTTOMRIGHT,
+            ]
+        ),
+        vol.Optional(ATTR_TRAVELTIME, default=DEFAULT_TRAVELTIME): cv.small_float,
+    }
+)
+
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up a Foscam IP Camera."""
-    add_entities([FoscamCam(config)])
+    _LOGGER.debug("Setting up the Foscam camera platform")
+
+    async def async_handle_ptz(service):
+        """Handle PTZ service call."""
+        movement = service.data.get(ATTR_MOVEMENT)
+        travel_time = service.data.get(ATTR_TRAVELTIME)
```

Comment:
Please use `dict[key]` for required schema keys and keys with default schema values.